src/plugins/interactive_setup/server/plugin.ts

```diff
@@ -65,45 +74,39 @@ export class UserSetupPlugin implements PrebootPlugin {
       })
     );
 
+    const elasticsearch = new ElasticsearchService(
+      core.elasticsearch,
+      this.#logger.get('elasticsearch')
+    );
+
     // If preliminary check above indicates that user didn't alter default Elasticsearch connection
     // details, it doesn't mean Elasticsearch connection isn't configured. There is a chance that they
     // already disabled security features in Elasticsearch and everything should work by default.
     // We should check if we can connect to Elasticsearch with default configuration to know if we
     // need to activate interactive setup. This check can take some time, so we should register our
     // routes to let interactive setup UI to handle user requests until the check is complete.
-    core.elasticsearch
-      .createClient('ping')
-      .asInternalUser.ping()
-      .then(
-        (pingResponse) => {
-          if (pingResponse.body) {
-            this.#logger.debug(
-              'Kibana is already properly configured to connect to Elasticsearch. Interactive setup mode will not be activated.'
-            );
-            this.#elasticsearchConnectionStatus = ElasticsearchConnectionStatus.Configured;
-            completeSetup({ shouldReloadConfig: false });
-          } else {
-            this.#logger.debug(
-              'Kibana is not properly configured to connect to Elasticsearch. Interactive setup mode will be activated.'
-            );
-            this.#elasticsearchConnectionStatus = ElasticsearchConnectionStatus.NotConfigured;
-          }
-        },
-        () => {
-          // TODO: we should probably react differently to different errors. 401 - credentials aren't correct, etc.
-          // Do we want to constantly ping ES if interactive mode UI isn't active? Just in case user runs Kibana and then
-          // configure Elasticsearch so that it can eventually connect to it without any configuration changes?
-          this.#elasticsearchConnectionStatus = ElasticsearchConnectionStatus.NotConfigured;
-        }
-      );
+    this.#connectionStatusSubscription = elasticsearch.connectionStatus$.subscribe((status) => {
+      if (status === ElasticsearchConnectionStatus.Configured) {
+        this.#logger.debug(
+          'Interactive setup mode is deactivated since Kibana is already properly configured to connect to Elasticsearch at http://localhost:9200.'
+        );
+        completeSetup({ shouldReloadConfig: false });
+      } else {
+        this.#logger.debug(
+          'Interactive setup mode is activated since Kibana cannot to connect to Elasticsearch at http://localhost:9200.'
```

Comment:
nit: Slightly shorter and easier to read "Starting interactive setup mode since ....."